.github/workflows/weekly_check.yml

```diff
@@ -0,0 +1,34 @@
+name: Weekly Stability Test
+on:
+  workflow_dispatch: # run on request (no need for PR)
+  schedule:
+    - cron: "0 9 * * 6" # every 9AM on Saturday
+defaults:
+  run:
+    shell: bash
+jobs:
+  weekly_stability_test:
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ['macos-11', 'ubuntu-20.04', 'windows-2019']
+        python-version: ['3.8']
+    name: stability test (${{ matrix.os }}, Python ${{ matrix.python-version }})
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Installing python
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Installing dependencies
+        run: |
+          pip install -e '.[default,tf,tfds]' pytest pytest-stress pytest-html
+      - name: Stability testing
+        run: |
+          pytest --html=stability_test_report.html --hours 8
```

Comment:
Is there any problem to run this long test on Github-hosted runner? e.g., billing.

Reply:
Haven't checked it yet. Then, until it is confirmed, CI will update it to 5 minutes for flow verification and I'll perform it manually for a while.

Reply:
it's free to use 2000 minutes per month for the public repository so I think it would be better to run on the stability tests on the self-hosted runner.

Reply:
Then shouldn't all periodically executed tests like nightly tests run in our self-hosted runner? It seems to hit the budget limit (2000 mins).

Reply:
I think so. but not sure whether it could be allowed to use gh runner with some billing plan.